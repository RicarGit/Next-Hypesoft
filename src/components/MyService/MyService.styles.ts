import styled from "styled-components"

export const MyServiceSection = styled.section`
  width: 100%;
  margin-top: 193px;
`

export const Container = styled.div`
  width: 1320px;
  margin: 0 auto;
  display: flex;
  flex-direction: column;
  align-items: center;
`

export const CardsContainer = styled.div`
  display: flex;
  gap: 24px;
`