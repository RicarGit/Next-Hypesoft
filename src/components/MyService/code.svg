<svg width="40" height="40" viewBox="0 0 40 40" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M26.6667 30L36.6667 20L26.6667 10" stroke="black" stroke-width="2.2" stroke-linecap="round" stroke-linejoin="round"/>
<path d="M13.3333 10L3.33325 20L13.3333 30" stroke="black" stroke-width="2.2" stroke-linecap="round" stroke-linejoin="round"/>
</svg>
