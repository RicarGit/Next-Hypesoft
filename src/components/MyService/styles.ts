import styled from "styled-components"

export const MyServiceSection = styled.section`
  position: absolute;
  top: 1945px;
  width: 100%;
  padding-bottom: 80px;
`

export const Container = styled.div`
  width: 1320px;
  margin: 0 auto;
  display: flex;
  flex-direction: column;
  align-items: center;
`

export const CardsContainer = styled.div`
  display: flex;
  gap: 24px;
`

export const Card = styled.div`
  width: 312px;
  height: 351px;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  padding: 32px;
  border-radius: 6px;
  gap: 100px;
  background-color: ${({ theme }) => theme.colors.white};
`

export const CardInfoContainer = styled.div`
  width: 248px;
  height: 147px;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: space-between;
`

export const CardInfo = styled.h2`
  position: relative;
  font-size: 32px;
  font-weight: 400;
  line-height: 40px;

  &::before {
    content: '';
    position: absolute;
    bottom: -16px;
    width: 64px;
    height: 3px;
    opacity: 30%;
    background-color: ${({ theme }) => theme.colors.secondary};
  }
`

export const CardActiveWrapper = styled.div`
  display: flex;
  justify-content: space-between;
  font-weight: 600;
  line-height: 24px;
  letter-spacing: 0.02em;
  text-transform: uppercase;
  gap: 8px;
  color: ${({ theme }) => theme.colors.secondary};
`