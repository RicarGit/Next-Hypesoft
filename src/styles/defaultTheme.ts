import { DefaultTheme } from "styled-components"

const defaultTheme: DefaultTheme = {
  colors: {
    primary: '#242424',
    secondary: '#ED1CA6',
    sec: '#7D00FF',
    white: '#FFFFFF',
    gray: {
      100: '#DFE3EB',
      500: '#A5A5A5',
    },
  },
  font: {
    family: 'Inter',
    weigths: {
      regular: 400,
      medium: 500,
      semiBold: 600,
      bold: 700
    },
  },
}

export default defaultTheme