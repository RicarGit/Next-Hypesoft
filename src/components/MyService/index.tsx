import * as S from './styles'

import Image from 'next/image'
import layersImage from './layers.svg'
import pinkBoxImage from './pink-box.svg'
import columnsImage from './columns.svg'
import codeImage from './code.svg'
import blackArrow from './black-arrow.svg'
import pinkArrow from 'components/shared/images/pink-arrow.svg'

import SectionHeader from 'components/shared/SectionHeader'

const MyService = () => {
  return (
    <S.MyServiceSection id='services'>
      <S.Container>
        <SectionHeader>My Service</SectionHeader>

        <S.CardsContainer>

          <S.Card>
            <Image src={layersImage} />
            <S.CardInfoContainer>
              <S.CardInfo>
                UI/UX <br />
                Design
              </S.CardInfo>
              <Image src={blackArrow} />
            </S.CardInfoContainer>
          </S.Card>

          <S.Card className='active'>
            <Image src={pinkBoxImage} />
            <S.CardInfoContainer>
              <S.CardInfo>
                Product <br />
                Design
              </S.CardInfo>
              <S.CardActiveWrapper>
                discuss now
                <Image src={pinkArrow} />
              </S.CardActiveWrapper>
            </S.CardInfoContainer>
          </S.Card>

          <S.Card>
            <Image src={columnsImage} />
            <S.CardInfoContainer>
              <S.CardInfo>
                Branding <br />
                Design
              </S.CardInfo>
              <Image src={blackArrow} />
            </S.CardInfoContainer>
          </S.Card>

          <S.Card>
            <Image src={codeImage} />
            <S.CardInfoContainer>
              <S.CardInfo>
                Front End <br />
                Development
              </S.CardInfo>
              <Image src={blackArrow} />
            </S.CardInfoContainer>
          </S.Card>

        </S.CardsContainer>

      </S.Container>
    </S.MyServiceSection>
  )
}

export default MyService