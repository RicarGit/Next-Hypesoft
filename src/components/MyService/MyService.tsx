import * as S from './MyService.styles'

import layersIcon from './layers.svg'
import pinkBoxIcon from './pink-box.svg'
import columnsIcon from './columns.svg'
import codeIcon from './code.svg'

import { SectionHeader } from '~/shared/SectionHeader/SectionHeader'
import { ServiceCard } from '~/components/ServiceCard'

export const MyService = () => {
  return (
    <S.MyServiceSection id='services'>
      <S.Container>
        <SectionHeader>My Service</SectionHeader>
        <S.CardsContainer>

          <ServiceCard
            icon={layersIcon}
            info1='UI/UX'
            info2='Design'
            alt='layers icon'
          />

          <ServiceCard
            icon={pinkBoxIcon}
            info1='Product'
            info2='Design'
            alt='pink box icon' className='active'
          />

          <ServiceCard
            icon={columnsIcon}
            info1='Branding'
            info2='Design'
            alt='columns icon'
          />

          <ServiceCard
            icon={codeIcon}
            info1='Front End'
            info2='Development'
            alt='code icon'
          />

        </S.CardsContainer>
      </S.Container>
    </S.MyServiceSection>
  )
}