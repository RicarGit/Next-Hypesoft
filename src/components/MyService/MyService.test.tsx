import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import defaultTheme from 'styles/defaultTheme'
import MyService from '.'

describe('MyService', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <ThemeProvider theme={defaultTheme}>
        <MyService />
      </ThemeProvider>
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})