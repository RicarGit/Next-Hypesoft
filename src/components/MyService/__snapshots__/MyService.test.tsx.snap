// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyService should match snapshot 1`] = `
<section
  class="sc-bczRLJ gfLvmB"
  id="services"
>
  <div
    class="sc-gsnTZi knAhub"
  >
    <div
      class="sc-hKMtZM iUPNYd"
    >
      <h2
        class="sc-eCYdqJ kwaGpO"
      >
        My Service
      </h2>
    </div>
    <div
      class="sc-dkzDqf dbvyyc"
    >
      <div
        class="sc-jSMfEi hjOdvB"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="layers icon"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
        <div
          class="sc-gKXOVf gymQTA"
        >
          <h2
            class="sc-iBkjds hypKw"
          >
            UI/UX
             
            <br />
            Design
          </h2>
          <div
            class="sc-ftvSup gbWuqU"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="black arrow"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
        </div>
      </div>
      <div
        class="sc-jSMfEi hjOdvB active"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="pink box icon"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
        <div
          class="sc-gKXOVf gymQTA"
        >
          <h2
            class="sc-iBkjds hypKw"
          >
            Product
             
            <br />
            Design
          </h2>
          <div
            class="sc-ftvSup gbWuqU"
          >
            Discuss Now
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="pink arrow"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
        </div>
      </div>
      <div
        class="sc-jSMfEi hjOdvB"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="columns icon"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
        <div
          class="sc-gKXOVf gymQTA"
        >
          <h2
            class="sc-iBkjds hypKw"
          >
            Branding
             
            <br />
            Design
          </h2>
          <div
            class="sc-ftvSup gbWuqU"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="black arrow"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
        </div>
      </div>
      <div
        class="sc-jSMfEi hjOdvB"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="code icon"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
        <div
          class="sc-gKXOVf gymQTA"
        >
          <h2
            class="sc-iBkjds hypKw"
          >
            Front End
             
            <br />
            Development
          </h2>
          <div
            class="sc-ftvSup gbWuqU"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="black arrow"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
